hal/src/main/native/athena/SerialPort.cpp

```diff
@@ -94,8 +94,20 @@ HAL_SerialPortHandle HAL_InitializeSerialPortDirect(HAL_SerialPort port,
   std::memset(&serialPort->tty, 0, sizeof(serialPort->tty));
 
   serialPort->baudRate = B9600;
-  cfsetospeed(&serialPort->tty, static_cast<speed_t>(serialPort->baudRate));
-  cfsetispeed(&serialPort->tty, static_cast<speed_t>(serialPort->baudRate));
+  if (cfsetospeed(&serialPort->tty,
+                  static_cast<speed_t>(serialPort->baudRate)) != 0) {
+    *status = errno;
```

Comment:
Do we want it to throw on error?  If so, I think these need to be -errno?

Reply:
Probably.